Source/JavaScriptCore/runtime/JSGenericTypedArrayViewPrototype.cpp

```diff
@@ -31,6 +31,45 @@
 
 namespace JSC {
 
+JSC_DEFINE_HOST_FUNCTION(uint8ArrayPrototypeSetFromHex, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSUint8Array* uint8Array = jsDynamicCast<JSUint8Array*>(callFrame->thisValue());
+    if (UNLIKELY(!uint8Array))
+        return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.setFromHex requires that |this| be a Uint8Array"_s);
+
+    IdempotentArrayBufferByteLengthGetter<std::memory_order_seq_cst> byteLengthGetter;
+    if (UNLIKELY(isIntegerIndexedObjectOutOfBounds(uint8Array, byteLengthGetter)))
+        return throwVMTypeError(globalObject, scope, typedArrayBufferHasBeenDetachedErrorMessage);
+
+    JSString* jsString = jsDynamicCast<JSString*>(callFrame->argument(0));
+    if (UNLIKELY(!jsString))
+        return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.setFromHex requires a string"_s);
+    if (UNLIKELY(jsString->length() % 2))
+        return JSValue::encode(throwSyntaxError(globalObject, scope, "Uint8Array.prototype.setFromHex requires a string of even length"_s));
+
+    StringView view = jsString->view(globalObject);
+    size_t read = 0;
+    size_t written = 0;
+    while (read < view.length() && written < uint8Array->length()) {
+        int tens = parseDigit(view[read++], 16);
+        int ones = parseDigit(view[read++], 16);
+        if (UNLIKELY(tens < 0 || tens > 255 || ones < 0 || ones > 255))
+            return JSValue::encode(throwSyntaxError(globalObject, scope, "Uint8Array.prototype.setFromHex requires a string containing only \"0123456789abcdefABCDEF\""_s));
+        uint8Array->typedVector()[written++] = (tens * 16) + ones;
+    }
+
+    JSObject* resultObject = constructEmptyObject(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    resultObject->putDirect(vm, vm.propertyNames->read, jsNumber(read));
+    resultObject->putDirect(vm, vm.propertyNames->written, jsNumber(written));
+
+    RELEASE_AND_RETURN(scope, JSValue::encode(resultObject));
```

Comment:
We do not need this.

Reply:
do you mind explaining when one would want to use `RELEASE_AND_RETURN`?